expression/aggregation/sum.go

```diff
@@ -76,9 +67,13 @@ func (sf *sumFunction) CalculateDefaultValue(schema *expression.Schema, ctx cont
 
 // GetType implements Aggregation interface.
 func (sf *sumFunction) GetType() *types.FieldType {
-	ft := types.NewFieldType(mysql.TypeNewDecimal)
+	var ft *types.FieldType
+	if sf.Args[0].GetType().Tp == mysql.TypeFloat || sf.Args[0].GetType().Tp == mysql.TypeDouble {
```

Comment:
this check can be extracted as IsTypeFloat() in types/etc.go ?

Reply:
I think it's ok